packages/terra-base/src/baseStyles.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * The following code is used to ensure global styles are loaded only once in an app
+ * that consumes terra components.
+ */
+if (!global.baseStylesLoaded) {
```

Comment:
Yes. This check is run in every terra-component that has styles. 

The first component which loads will run this condition and it will evaluate that `global.baseStylesLoaded` is `false`. The `!` operator will cast `global.baseStylesLoaded` to `true` causing the code inside the if condition to run.

Then on line 13, we set `global.baseStylesLoaded` to `true`. The next terra component which loads will run this condition and it will evaluate that `global.baseStylesLoaded` is `true`. The `!` operator will cast `global.baseStylesLoaded` to `false` causing the code inside the if condition to be skipped preventing the duplication of the global base styles.